### PR TITLE
automation - run tests on multiple OS and Docker version

### PR DESCRIPTION
**Pull request by jiaqiluo**
Issue: https://github.com/rancher/rancher/issues/24722

Add a pipeline file that can 
- run Rancher server
- provision clusters with different OS and docker version
- run tests on those clusters

Add the following two private images:

- RHEL7.7_Native_Docker_1.13.1
- RHEL7.7_Upstream_Docker_19.03

### CODE REVIEW

**Comment by bmdepesa** [comment]
Can you squash the commits @jiaqiluo ?

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_custom_cluster_matrix [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        }
+
+        stage('install Rancher') {
+          // skip provisioning a new Rancher Server 
+          // if CATTLE_TEST_URL, ADMIN_TOKEN, USER_TOKEN are provided
+          if(env.CATTLE_TEST_URL != "" || env.ADMIN_TOKEN != "" || env.USER_TOKEN != "") {
+            println "Rancher Server is provided: ${env.CATTLE_TEST_URL}"
+          }
+          else {
+            def racnherContainerName = "${JOB_NAME}${env.BUILD_NUMBER}_rancher_server"
```
[the remark this answers is not in the document]
Comment (jiaqiluo) [reply tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_custom_cluster_matrix]: fixed

**Comment by bmdepesa** [comment]
LGTM

**Comment by sangeethah** [comment]
LGTM

**Pull request merged**